### PR TITLE
Add support for `android:gravity`

### LINKED ISSUE #4: Support gravity attribute
You can see that the width of TickerView does a weird jump when you have the width set to `wrap_content` and you go from a longer string, such as "August" to a shorter string, such as "July". It will appear normal at first, with the "July" text appearing centered within within the view. When you then set the text to a similarly short string, such as "June", you will see the text jump from being center aligned to being left aligned. 

I am going to investigate this further and see if I can find a solution.

### PR DESCRIPTION
**Pull request by jinatonic**
Add `gravity` support in `TickerView` which defaults to `Gravity.START`. Add corresponding test to check for gravity realignment. Modified the sample project to use `Gravity.CENTER`.

fixes #4

### CODE REVIEW

File: ticker/src/main/java/com/robinhood/ticker/TickerView.java [hunk 1]
```diff
@@ -229,9 +233,11 @@ public int getTextColor() {
      * @param color the color to set the text to.
      */
     public void setTextColor(int color) {
-        textColor = color;
-        textPaint.setColor(textColor);
-        invalidate();
+        if (this.textColor != color) {
+            textColor = color;
+            textPaint.setColor(textColor);
+            invalidate();
```
Comment (danh32) [inline ticker/src/main/java/com/robinhood/ticker/TickerView.java]: call `onTextPaintChanged()` instead?

File: ticker/src/main/java/com/robinhood/ticker/TickerView.java [hunk 2]
```diff
@@ -399,4 +418,41 @@ protected void onDraw(Canvas canvas) {
...
     }
+
+    private void realignAndClipCanvasForGravity(Canvas canvas) {
+        final float currentWidth = columnManager.getCurrentWidth();
+        final float currentHeight = metrics.getCharHeight();
+        realignAndClipCanvasForGravity(canvas, gravity, canvasFrame, currentWidth, currentHeight);
+    }
+
+    // VisibleForTesting
+    static void realignAndClipCanvasForGravity(Canvas canvas, int gravity, Rect canvasFrame,
```
Comment (danh32) [inline ticker/src/main/java/com/robinhood/ticker/TickerView.java]: should we rename `canvasFrame` to `viewBounds`?

File: ticker/src/main/java/com/robinhood/ticker/TickerView.java [hunk 3]
```diff
@@ -380,17 +409,7 @@ protected void onDraw(Canvas canvas) {
...
-        final float currentWidth = columnManager.getCurrentWidth();
-        final float currentHeight = metrics.getCharHeight();
-
-        // We translate first by the canvas frame and then re-align so that we are drawing in
-        // the center of the canvas (if availableWidth is greater than currentWidth).
-        final float translationX = canvasFrame.left + (availableWidth - currentWidth) / 2f;
-        final float translationY = canvasFrame.top + (availableHeight - currentHeight) / 2f;
-        canvas.translate(translationX ,translationY);
-        canvas.clipRect(0f, 0f, currentWidth, currentHeight);
+        realignAndClipCanvasForGravity(canvas);
```
Comment (danh32) [inline ticker/src/main/java/com/robinhood/ticker/TickerView.java]: slick!

File: ticker/src/main/java/com/robinhood/ticker/TickerView.java (hunk 1 above)
Comment (jinatonic) [inline ticker/src/main/java/com/robinhood/ticker/TickerView.java]: I realized that `onTextPaintChanged` was a poorly named method. It should really be `onTextPaintMeasurementChanged`. I updated the method's name and left this as `invalidate()`.

**Pull request merged**